Source/themes.cpp

```diff
@@ -181,16 +182,22 @@ bool TFit_GoatShrine(int t)
 bool CheckThemeObj3(int xp, int yp, int t, int f)
 {
 	for (int i = 0; i < 9; i++) {
-		if (xp + trm3x[i] < 0 || yp + trm3y[i] < 0)
+		Point testPosition = Point { xp, yp } + Displacement { trm3x[i], trm3y[i] };

```

Comment:
`trm3x`/`trm3y` are just begging to become an array of directions :)

Reply:
That was the original approach in #3336 :D (then I overcomplicated it with the rectangle iterator, I'll get that working eventually)